.github/workflows/release.yml

```diff
@@ -349,18 +257,18 @@ jobs:
       - name: Pull self-host image
         run: |
           if [[ "${{ github.event.inputs.release_type }}" == "Dry Run" ]]; then
-            docker pull bitwarden/self-host:dev
+            docker pull $_AZ_REGISTRY/self-host:dev
           else
-            docker pull bitwarden/self-host:beta
+            docker pull $_AZ_REGISTRY/self-host:beta
           fi
```

Comment:
We need to log in to the prod registry before this step.  The steps in this job need to be moved around.